From 62290a5c194b30a096cbcc6bb8680a5a1e627714 Mon Sep 17 00:00:00 2001
From: Jerry Hoemann <jerry.hoemann@hpe.com>
Date: Thu, 3 May 2018 15:00:55 -0600
Subject: [PATCH] watchdog: hpwdt: Claim NMIs generated by iLO5
Git-commit: 62290a5c194b30a096cbcc6bb8680a5a1e627714
Patch-mainline: v4.18-rc1
References: FATE#325978,bsc#1105831

On iLO5 going forward we want to return and not claim the NMI, if
the NMI was NOT gnerated by the iLO as a result of the watchdog
timing out or an explicit generate NMI.

The sense of the test in is inverted and prevents hpwdt_pretimeout
from claiming NMIs when it should.

Signed-off-by: Jerry Hoemann <jerry.hoemann@hpe.com>
Fixes: a042229a18ac ("watchdog: hpwdt: Update nmi_panic message.")
Reviewed-by: Guenter Roeck <linux@roeck-us.net>
Signed-off-by: Guenter Roeck <linux@roeck-us.net>
Signed-off-by: Wim Van Sebroeck <wim@iguana.be>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/watchdog/hpwdt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/watchdog/hpwdt.c b/drivers/watchdog/hpwdt.c
index a43ab2cecca2..9dc62a461451 100644
--- a/drivers/watchdog/hpwdt.c
+++ b/drivers/watchdog/hpwdt.c
@@ -159,7 +159,7 @@ static int hpwdt_pretimeout(unsigned int ulReason, struct pt_regs *regs)
 		"3. OA Forward Progress Log\n"
 		"4. iLO Event Log";
 
-	if (ilo5 && ulReason == NMI_UNKNOWN && mynmi)
+	if (ilo5 && ulReason == NMI_UNKNOWN && !mynmi)
 		return NMI_DONE;
 
 	if (ilo5 && !pretimeout)
-- 
2.18.0

